### PR TITLE
[SPARK-8471][ML]Rename DiscreteCosineTransformer to DCT

### PR DESCRIPTION
**Pull request by feynmanliang**
Rename DiscreteCosineTransformer and related classes to DCT.

### CODE REVIEW

**Comment by feynmanliang** [comment]
Github's diff against master doesn't seem to be picking up `DiscreteCosineTransformer`

**Comment by jkbradley** [comment]
Was this branch created from master, or from your old PR branch?  The former is better.

**Comment by feynmanliang** [comment]
From old PR branch; I just merged it with master and the diffs are correct now.

File: mllib/src/test/java/org/apache/spark/ml/feature/JavaDCTSuite.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public void javaCompatibilityTest() {
     double[] expectedResult = input.clone();
     (new DoubleDCT_1D(input.length)).forward(expectedResult, true);
 
-    DiscreteCosineTransformer DCT = new DiscreteCosineTransformer()
+    DCT DCT = new DCT()
```
Comment (jkbradley) [inline mllib/src/test/java/org/apache/spark/ml/feature/JavaDCTSuite.java]: You might rename this value to "dct"

**Comment by jkbradley** [comment]
Other than that, it looks good to me

**Comment by jkbradley** [comment]
LGTM, merging into master
Thanks!

**Pull request closed without merging**